vernac/vernacexpr.ml

```diff
@@ -477,10 +482,15 @@ type nonrec vernac_expr =
   | VernacShow of showable
   | VernacCheckGuard
   | VernacProof of Genarg.raw_generic_argument option * section_subset_expr option
-  | VernacProofMode of string
 
-  (* For extension *)
-  | VernacExtend of extend_name * Genarg.raw_generic_argument list
+  | VernacAddOption of Goptions.option_name * Goptions.table_value list
+  | VernacRemoveOption of Goptions.option_name * Goptions.table_value list
+
+type 'a vernac_expr_gen =
+  | VernacSynterp of 'a
+  | VernacPure of pure_vernac_expr
```

Comment:
I don't think this name is correct, it is not generally pure but only pure wrt syntax. Maybe 
```suggestion
  | VernacSynPure of synpure_vernac_expr
```
?

Reply:
It's not about anticipating, it's about correctness today

Reply:
Today we distinguish only syntax effects at that place. So pure = no syntax effect.

Reply:
Name changed.